Source/Private/LevelStatsCollector.cpp

```diff
@@ -1,22 +1,67 @@
 ﻿#include "LevelStatsCollector.h"
 
+#include "LevelStatsCollectorState.h"
+
 #include <Components/SceneCaptureComponent2D.h>
+#include <Engine/Engine.h>
 #include <Engine/LevelBounds.h>
 #include <Engine/TextureRenderTarget2D.h>
 #include <ImageUtils.h>
 
+DEFINE_LOG_CATEGORY( LogLevelStatsCollector );
+
+FCustomPerformanceChart::FCustomPerformanceChart( const FDateTime & start_time, FStringView chart_label, FStringView output_path ) :
+    FPerformanceTrackingChart( start_time, FString( chart_label ) ),
+    CustomOutputPath( output_path )
+{
+}
+
+void FCustomPerformanceChart::DumpFPSChartToCustomLocation( FStringView map_name )
+{
+    TArray< const FPerformanceTrackingChart * > charts;
+    charts.Add( this );
+
+    DumpChartsToOutputLog( AccumulatedChartTime, charts, FString( map_name ) );
+
+#if ALLOW_DEBUG_FILES
+    IFileManager::Get().MakeDirectory( *CustomOutputPath, true );
+
+    {
+        const auto log_filename = CustomOutputPath / CreateFileNameForChart( TEXT( "FPS" ), map_name, TEXT( ".log" ) );
+        DumpChartsToLogFile( AccumulatedChartTime, charts, FString( map_name ), log_filename );
+    }
+
+    {
+        const auto map_and_chart_label = ChartLabel.IsEmpty()
+                                             ? FString( map_name )
+                                             : FString::Printf( TEXT( "%s-%s" ), *ChartLabel, *FString( map_name ) );
+        const auto html_filename = CustomOutputPath / CreateFileNameForChart(
+                                                          TEXT( "FPS" ),
+                                                          map_name,
+                                                          FString::Printf( TEXT( "-%s.html" ), *CaptureStartTime.ToString() ) );
+        DumpChartsToHTML( AccumulatedChartTime, charts, map_and_chart_label, html_filename );
+    }
+#endif
+}
+
+FString FCustomPerformanceChart::CreateFileNameForChart( FStringView, FStringView, FStringView file_extension )
```

Comment:
you're not using the first parameter, just drop it

Reply:
done

Reply:
So this will become totally irrelevant next PR (I'm not using these functions anymore)